components/Discussion/DiscussionProvider/hooks/useOverlay.ts

```diff
@@ -0,0 +1,37 @@
+import { useState } from 'react'
+
+export type OverlayState<T> = {
+  data: T | null
+  open: boolean
+  setOpen: (open: boolean) => void
+  handleOpen: (data?: T | null) => void
+  handleClose: () => void
+}
+
+/**
+ * Hook to handle the state of an overlay as well as data passed to it.
+ */
+function useOverlay<T>(): OverlayState<T> {
+  const [open, setOpen] = useState(false)
```

Comment:
This state is unnecessary since this is always true: `open === !!data`. And a custom hook for just one data state seems totally overkill to me. IMO a better abstraction would be that the discussion provider has a method to set a share url and one to trigger the preference overlay (already present in actions, had to add it to work correctly in reply composer)